src/map/script.c

```diff
@@ -17084,6 +17084,1251 @@ BUILDIN(getunittype) {
+			ShowWarning("buildin_setunitdata: Character ID %d not found for master change!\n",val);
+			script_pushint(st, 0);
+			return false;
+		}
+		break;
+	case UDT_MAPIDXY:
+		if ((val = map->mapname2mapid(mapname)) == -1) {
+			ShowError("buildin_setunitdata: Non-existent map %s provided.\n", mapname);
+			return false;
+		}
+		setunitdata_check_int(5); 
+		setunitdata_check_int(6);
+		setunitdata_check_bounds(5, 0, MAX_MAP_SIZE/2);
+		setunitdata_check_bounds(6, 0, MAX_MAP_SIZE/2);
+		val2 = script_getnum(st, 5);
+		val3 = script_getnum(st, 6);
+		break;
+		// fall through
+	case UDT_WALKTOXY:
+		setunitdata_assert_arg(6, false);
+		setunitdata_check_int(5);
+		val2 = script_getnum(st, 5);
+		setunitdata_check_bounds(4, 0, MAX_MAP_SIZE/2);
+		setunitdata_check_bounds(5, 0, MAX_MAP_SIZE/2);
+		break;
+	case UDT_SPEED:
+		setunitdata_check_bounds(4, 0, MAX_WALK_SPEED);
+		break;
+	case UDT_MODE:
+		setunitdata_check_bounds(4, MD_NONE, MD_MASK);
+		break;
+	case UDT_AI:
+		setunitdata_check_bounds(4, AI_NONE, AI_MAX-1);
+		break;
+	case UDT_SCOPTION:
+		setunitdata_check_bounds(4, OPTION_NOTHING, OPTION_COSTUME);
+		break;
+	case UDT_SEX:
+		setunitdata_check_bounds(4, SEX_FEMALE, SEX_MALE);
+		break;
+	case UDT_HAIRSTYLE:
+		setunitdata_check_bounds(4, 0, battle->bc->max_hair_style);
+		break;
+	case UDT_HAIRCOLOR:
+		setunitdata_check_bounds(4, 0, battle->bc->max_hair_color);
+		break;
+	case UDT_LOOKDIR:
+		setunitdata_check_bounds(4, 0, 7);
+		break;
+	case UDT_CANMOVETICK:
+		setunitdata_check_uint64(4, 0, UINT64_MAX);
+		break;
+	case UDT_STR:
+	case UDT_AGI:
+	case UDT_VIT:
+	case UDT_INT:
+	case UDT_DEX:
+	case UDT_LUK:
+	case UDT_STATPOINT:
+	case UDT_ATKRANGE:
+	case UDT_ATKMIN:
+	case UDT_ATKMAX:
+	case UDT_MATKMIN:
+	case UDT_MATKMAX:
+	case UDT_AMOTION:
+	case UDT_ADELAY:
+	case UDT_DMOTION:
+	setunitdata_check_bounds(4, 0, USHRT_MAX);
+		break;
+	case UDT_DEF:
+	case UDT_MDEF:
+	case UDT_HIT:
+	case UDT_FLEE:
+	case UDT_PDODGE:
+	case UDT_CRIT:
+		setunitdata_check_bounds(4, 0, SHRT_MAX);
+		break;
+	case UDT_HUNGER:
+		setunitdata_check_bounds(4, 0, 99);
+		break;
+	case UDT_RACE:
+	case UDT_ELETYPE:
+	case UDT_ELELEVEL:
+		setunitdata_check_bounds(4, 0, CHAR_MAX);
+		break;
+	default:
+		break;
+	}
+
+#undef setunitdata_check_bounds
+#undef setunitdata_check_uint64
+#undef setunitdata_assert_arg
+#undef setunitdata_check_int
+#undef setunitdata_check_string
+
+	/* Set the values */
+	switch (bl->type) {
+	case BL_MOB:
+	{
+		struct mob_data *md = BL_UCAST(BL_MOB, bl);
+		nullpo_retr(false, md);
+		
+		switch (type)
+		{
+		case UDT_SIZE:        md->status.size = (unsigned char) val; break;
+		case UDT_LEVEL:       md->level = val; break;
+		case UDT_HP:
+				status->set_hp(bl, (unsigned int) val, 0);
+				clif->charnameack(0, &md->bl);
+				break;
+		case UDT_MAXHP:
+				md->status.max_hp = (unsigned int) val;
+				clif->charnameack(0, &md->bl);
+				break;
+		case UDT_SP:          status->set_sp(bl, (unsigned int) val, 0); break;
```

Comment:
Can you clean up the style of this switch to make it more readable? no code on the same line as the case labels (doesn't look that great when there are single-line and multi-line cases), only one level of indentation